### PR TITLE
Correct the usage of pointers of database objects

### PR DESCRIPTION
**Pull request by Ishad-M-I-M**
Consider only after merging #224

### CODE REVIEW

File: src/performance/metrics/PerformanceUtil.h [hunk 1]
```diff
@@ -84,8 +84,8 @@ class PerformanceUtil {
                                            std::string category, int elapsedTime);
 
  private:
-    // static SQLiteDBInterface sqlLiteDB;
-    // static PerformanceSQLiteDBInterface perfDb;
+    // static SQLiteDBInterface *sqlLiteDB;
```
Comment (miyurud) [inline src/performance/metrics/PerformanceUtil.h:87]: Since these two lines are commented, lets delete those.

Comment (Ishad-M-I-M) [inline src/performance/metrics/PerformanceUtil.h:87]:
```suggestion
```

File: src/performance/metrics/PerformanceUtil.h [hunk 2]
```diff
@@ -84,8 +84,8 @@ class PerformanceUtil {
...
+    // static PerformanceSQLiteDBInterface *perfDb;
```
Comment (Ishad-M-I-M) [inline src/performance/metrics/PerformanceUtil.h:88]:
```suggestion
```

**Review by miyurud — CHANGES_REQUESTED** [review]
Please address the PR comment

File: src/server/JasmineGraphServer.cpp [hunk 3]
```diff
@@ -65,33 +74,19 @@ JasmineGraphServer *JasmineGraphServer::getInstance() {
...
-JasmineGraphServer::JasmineGraphServer() {}
-
 JasmineGraphServer::~JasmineGraphServer() {
-    puts("Freeing up server resources.");
+    server_logger.info("Freeing up server resources.");
     sqlite->finalize();
+    performanceSqlite->finalize();
+    delete sqlite;
+    delete performanceSqlite;
+    delete jobScheduler;
```
Comment (thevindu-w) [inline src/server/JasmineGraphServer.cpp:83]: Why not `this->sqlite` and other variables (with `this->`)?

File: src/performance/metrics/PerformanceUtil.cpp [hunk 4]
```diff
@@ -17,12 +17,14 @@ using namespace std::chrono;
...
-SQLiteDBInterface sqlLiteDB = *new SQLiteDBInterface();
-PerformanceSQLiteDBInterface perfDb = *new PerformanceSQLiteDBInterface();
+SQLiteDBInterface *sqlLiteDB;
+PerformanceSQLiteDBInterface *perfDb;
 
 void PerformanceUtil::init() {
-    sqlLiteDB.init();
-    perfDb.init();
+    sqlLiteDB = new SQLiteDBInterface();
+    perfDb = new PerformanceSQLiteDBInterface();
```
Comment (thevindu-w) [inline src/performance/metrics/PerformanceUtil.cpp:25]: If `PerformanceUtil::init()` is called multiple times, won't this cause a memory leak?

File: src/frontend/JasmineGraphFrontEnd.cpp [hunk 5]
```diff
@@ -757,8 +757,9 @@ static void add_graph_command(std::string masterIP, int connFd, SQLiteDBInterfac
             "vertexcount,centralpartitioncount,edgecount) VALUES(\"" +
             name + "\", \"" + path + "\", \"" + uploadStartTime + "\", \"\",\"" +
             to_string(Conts::GRAPH_STATUS::LOADING) + "\", \"\", \"\", \"\")";
-        int newGraphID = sqlite.runInsert(sqlStatement);
-        MetisPartitioner *partitioner = new MetisPartitioner(&sqlite);
+        int newGraphID = sqlite->runInsert(sqlStatement);
+        JasmineGraphServer *jasmineServer = JasmineGraphServer::getInstance();
```
Comment (thevindu-w) [inline src/frontend/JasmineGraphFrontEnd.cpp:761]: Is this variable `jasmineServer` used?

File: src/frontend/JasmineGraphFrontEnd.cpp [hunk 6]
```diff
@@ -931,8 +932,9 @@ static void add_graph_cust_command(std::string masterIP, int connFd, SQLiteDBInt
             "vertexcount,centralpartitioncount,edgecount) VALUES(\"" +
             name + "\", \"" + edgeListPath + "\", \"" + uploadStartTime + "\", \"\",\"" +
             to_string(Conts::GRAPH_STATUS::LOADING) + "\", \"\", \"\", \"\")";
-        int newGraphID = sqlite.runInsert(sqlStatement);
-        MetisPartitioner *partitioner = new MetisPartitioner(&sqlite);
+        int newGraphID = sqlite->runInsert(sqlStatement);
+        JasmineGraphServer *jasmineServer = JasmineGraphServer::getInstance();
```
Comment (thevindu-w) [inline src/frontend/JasmineGraphFrontEnd.cpp:936]: Is this variable `jasmineServer` used?

**Review by miyurud — APPROVED** [review]

**Pull request merged**